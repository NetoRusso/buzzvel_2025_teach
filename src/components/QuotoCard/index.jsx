
import styles from "./styles.module.css";
import Image from "next/image";

import squares from "@/assets/SVG/squares.svg";
import chat_quotes from "@/assets/SVG/chat_quote.svg";
import arrow_left from "@/assets/SVG/arrow_left_white.svg";
import arrow_right from "@/assets/SVG/arrow_right_blue.svg";
import photo from "@/assets/picture_quote.png";


const QuotoCard = () => {

  return (
    <section className={styles.container} aria-label="Quotes slider">

      <div className={styles.content}>

        <div className={styles.main_container}>

          <div className={styles.slider}>
            <div className={styles.box_1}>
              <div className={styles.btn}>
                <Image src={arrow_left} alt="arrow left" width={18} height={31.5} loading="lazy" />
              </div>
              <div className={styles.box_1_content}>
                <p className={styles.texts_msg}>
                  Id urna, nisl, ut quam. Diam suspendisse fringilla quam arcu mattis est velit in. Nibh in purus sit convallis phasellus ut. At vel erat ultricies commodo. Neque suspendisse a habitasse commodo.
                </p>
                <div className={styles.texts_ass}>
                  <p className={styles.texts_ass_name}>
                    Marie Poirot,
                  </p>
                  <p className={styles.texts_ass_role}>
                    Bigapp
                  </p>
                </div>
                <div className={styles.dots} />
              </div>
            </div>
            <div className={styles.box_2}>
              <div className={styles.box_2_content}>
                <Image src={photo} alt="photo" width={528} height={415} className={styles.content_photo} loading="lazy" />
              </div>
              <div className={styles.btn}>
                <Image src={arrow_right} alt="arrow right" width={18} height={31.5} loading="lazy" />
              </div>
              <div className={styles.details_bg} />
              <Image src={chat_quotes} alt="chat quotes" width={48} height={48} className={styles.details_chat} loading="lazy" />
              <Image src={squares} alt="squares" width={155} height={155} className={styles.details_squares} loading="lazy" />
            </div>
          </div>

        </div>
      </div>
    </section>
  )
};

export default QuotoCard;