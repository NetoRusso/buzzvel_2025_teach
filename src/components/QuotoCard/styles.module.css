
.container {
  width: 100%;
  height: auto;
  min-height: 100vh;
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
  padding-bottom: 64px;
  overflow-y: clip;
}

.content { 
  width: 100%;
  height: auto;
  background-color: var(--yellow-600);
  display: flex;
  align-items: center;
  justify-content: center;
}

.main_container { 
  width: 100%;
  max-width: 1440px;
  height: auto;
  min-height: 518px;
  display: flex;
  align-items: center;
  justify-content: center;
  overflow-y: visible;
}

.slider {
  width: 100%;
  max-width: 1280px;
  height: auto;
  min-height: 415px;
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: space-between;
  gap: 40px;
  position: relative;

}

.box_1 { 
  width: 45%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  gap: 16px;
}

.btn {
  width: 48px;
  height: 48px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.box_1_content {
  width: calc(234.857142857px + 20.357142857vw);
  max-width: 528px;
  min-width: 294px;
  height: auto;
  max-height: 325px;
  min-height: 282px;
  display: flex;
  flex-flow: column nowrap;
  align-items: flex-start;
  justify-content: center;
  gap: 24px;
}

.texts_msg {
  font-size: clamp(1.125rem, 0.9842rem + 0.5731vw, 1.5rem);
  line-height: 160%;
  font-weight: 400;
  color: var(--blue-Gray-900);
}

.texts_ass_name { 
  font-size: clamp(1.125rem, 0.9842rem + 0.5731vw, 1.5rem);
  line-height: 160%;
  font-weight: 400;
  color: var(--blue-Gray-900);
}

.texts_ass_role { 
  font-size: clamp(1.125rem, 0.9842rem + 0.5731vw, 1.5rem);
  line-height: 140%;
  font-weight: 700;
  color: var(--blue-Gray-900);
}

.dots { 
  width: 32px;
  height: 8px;
  background-color: var(--blue-Gray-900);
  border-radius: 4px;
  position: relative;
  margin-top: 16px;
}

.dots::before {
  content: '';
  width: 8px;
  height: 8px;
  background-color: var(--background-white);
  border-radius: 4px;
  position: absolute;
  top: 0;
  right: -16px;
  box-shadow: 16px 0 0 0 var(--background-white),
  32px 0 0 0 var(--background-white),
  48px 0 0 0 var(--background-white);
}

.dots::after {
  content: '';
  width: 32px;
  height: 8px;
  background-color: var(--blue-Gray-900);
  border-radius: 4px;
  position: absolute;
  top: 0;
  left: 0;
}

.box_2 {
  width: 50%;
  min-width: 300px;
  height: auto;
  min-height: 294px;
  max-height: 415px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-end;
  gap: 16px;
  position: relative;

}

.box_2_content {
  width: 80%;
  max-width: 528px;
  min-width: 300px;
  height: auto;
  min-height: 294px;
  max-height: 415px;
  position: relative;
  
}

.content_photo { 
  width: 100%;
  height: auto;
  position: absolute;
  z-index: 1;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.details_bg { 
  width: calc(290px + 7vw);
  height: calc(325px + 13vw);
  background-color: var(--blue-Gray-900);
  position: absolute;
  border-radius: 50px;
  top: 50%;
  left: 45%;
  transform: translate(-50%, -50%) rotate(-30deg);
  z-index: 0;
}

.details_squares { 
  position: absolute;
  z-index: 2;
  bottom: -30%;
  left: 20%;

}

.details_chat {
  position: absolute;
  z-index: 2;
  top: -48px;
  right: 16px;
} 

@media screen and (max-width: 960px) {
  .content{
    padding: 24px 0;
  }
  .btn {
    display: none;
  }
  .slider {
    flex-direction: column;
    align-items: center;
    justify-content: center;
    gap: 80px;
  }
  .box_1 {
    width: 100%;
    margin-bottom: 24px;
  }
}


