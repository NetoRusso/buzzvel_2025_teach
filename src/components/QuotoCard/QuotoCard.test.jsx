import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import QuotoCard from './index';

jest.mock('next/image', () => {
  const ImageComponent = (props) => {
    let { src: srcProp, alt, ...rest } = props;
    let finalSrc = srcProp;
    if (typeof srcProp === 'object' && srcProp !== null && srcProp.src) {
      finalSrc = srcProp.src;
    }
    return <img src={finalSrc} alt={alt || ''} {...rest} />;
  };
  ImageComponent.displayName = 'NextImageMock';
  return {
    __esModule: true,
    default: ImageComponent,
  };
});

describe('QuotoCard Component', () => {
  test('renders static text content correctly', () => {
    render(<QuotoCard />);
    expect(screen.getByText(/Id urna, nisl, ut quam\. Diam suspendisse fringilla quam arcu mattis est velit in\./i)).toBeInTheDocument();
    expect(screen.getByText(/Marie Poirot,/i)).toBeInTheDocument();
    expect(screen.getByText(/Bigapp/i)).toBeInTheDocument();
  });

  test('renders all images with correct alt texts and src attributes', () => {
    render(<QuotoCard />);
    const expectedImageSrc = '/img.jpg';

    const arrowLeft = screen.getByAltText('arrow left');
    expect(arrowLeft).toBeInTheDocument();
    expect(arrowLeft).toHaveAttribute('src', expectedImageSrc);
    expect(arrowLeft).toHaveAttribute('width', '18');

    const arrowRight = screen.getByAltText('arrow right');
    expect(arrowRight).toBeInTheDocument();
    expect(arrowRight).toHaveAttribute('src', expectedImageSrc);
    expect(arrowRight).toHaveAttribute('width', '18');

    const photo = screen.getByAltText('photo');
    expect(photo).toBeInTheDocument();
    expect(photo).toHaveAttribute('src', expectedImageSrc);
    expect(photo).toHaveAttribute('width', '528');

    const chatQuotes = screen.getByAltText('chat quotes');
    expect(chatQuotes).toBeInTheDocument();
    expect(chatQuotes).toHaveAttribute('src', expectedImageSrc);
    expect(chatQuotes).toHaveAttribute('width', '48');

    const squares = screen.getByAltText('squares');
    expect(squares).toBeInTheDocument();
    expect(squares).toHaveAttribute('src', expectedImageSrc);
    expect(squares).toHaveAttribute('width', '155');
  });

  test('renders decorative elements', () => {
    render(<QuotoCard />);
  });

  test('has correct aria-label for accessibility', () => {
    render(<QuotoCard />);
    expect(screen.getByRole('region', { name: 'Quotes slider' })).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = render(<QuotoCard />);
    expect(container.firstChild).toMatchSnapshot();
  });
});