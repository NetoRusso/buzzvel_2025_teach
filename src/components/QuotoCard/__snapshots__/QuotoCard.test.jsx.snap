// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuotoCard Component matches snapshot 1`] = `
<section
  aria-label="Quotes slider"
  class="container"
>
  <div
    class="content"
  >
    <div
      class="main_container"
    >
      <div
        class="slider"
      >
        <div
          class="box_1"
        >
          <div
            class="btn"
          >
            <img
              alt="arrow left"
              height="31.5"
              loading="lazy"
              src="/img.jpg"
              width="18"
            />
          </div>
          <div
            class="box_1_content"
          >
            <p
              class="texts_msg"
            >
              Id urna, nisl, ut quam. Diam suspendisse fringilla quam arcu mattis est velit in. Nibh in purus sit convallis phasellus ut. At vel erat ultricies commodo. Neque suspendisse a habitasse commodo.
            </p>
            <div
              class="texts_ass"
            >
              <p
                class="texts_ass_name"
              >
                Marie Poirot,
              </p>
              <p
                class="texts_ass_role"
              >
                Bigapp
              </p>
            </div>
            <div
              class="dots"
            />
          </div>
        </div>
        <div
          class="box_2"
        >
          <div
            class="box_2_content"
          >
            <img
              alt="photo"
              class="content_photo"
              height="415"
              loading="lazy"
              src="/img.jpg"
              width="528"
            />
          </div>
          <div
            class="btn"
          >
            <img
              alt="arrow right"
              height="31.5"
              loading="lazy"
              src="/img.jpg"
              width="18"
            />
          </div>
          <div
            class="details_bg"
          />
          <img
            alt="chat quotes"
            class="details_chat"
            height="48"
            loading="lazy"
            src="/img.jpg"
            width="48"
          />
          <img
            alt="squares"
            class="details_squares"
            height="155"
            loading="lazy"
            src="/img.jpg"
            width="155"
          />
        </div>
      </div>
    </div>
  </div>
</section>
`;
